cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -151,10 +151,14 @@ private boolean validateSecuredInterception(ChannelHandlerContext ctx, HttpReque
         LOG.debug("Authentication failed due to invalid token, reason={};", tokenState);
       }
       JsonArray externalAuthenticationURIs = new JsonArray();
-
-      //Waiting for service to get discovered
-      stopWatchWait(externalAuthenticationURIs);
-
+      String announceAddress = configuration.get(Constants.Security.AUTH_SERVER_ANNOUNCE_ADDRESS);
+      // If the announceAddress is set in configuration, return it directly
+      if (announceAddress != null) {
+        externalAuthenticationURIs.add(new JsonPrimitive(announceAddress));
```

Comment:
I believe the `AUTH_SERVER_ANNOUNCE_ADDRESS` setting is not a URI (I believe it's just a hostname) and the `externalAuthenticationURIs` should have URI as it's being used as the response. Also, isn't the idea is to having a different URI (mainly a different hostname) for this response (hence for the client to use) then the one used by the auth server to bind to and announce itself (which is governed by the `AUTH_SERVER_ANNOUNCE_ADDRESS` config).